manim/mobject/text/numbers.py

```diff
@@ -83,38 +94,38 @@ def construct(self):
 
     def __init__(
         self,
-        number: float = 0,
+        number: float | complex = 0,
         num_decimal_places: int = 2,
-        mob_class: VMobject = MathTex,
+        mob_class: type[TextLike] = MathTex,
         include_sign: bool = False,
         group_with_commas: bool = True,
         digit_buff_per_font_unit: float = 0.001,
         show_ellipsis: bool = False,
         unit: str | None = None,  # Aligned to bottom unless it starts with "^"
         unit_buff_per_font_unit: float = 0,
         include_background_rectangle: bool = False,
-        edge_to_fix: Sequence[float] = LEFT,
+        edge_to_fix: Vector3D = LEFT,
         font_size: float = DEFAULT_FONT_SIZE,
         stroke_width: float = 0,
         fill_opacity: float = 1.0,
-        **kwargs,
-    ):
+        **kwargs: Any,
+    ) -> None:
         super().__init__(**kwargs, stroke_width=stroke_width)
-        self.number = number
-        self.num_decimal_places = num_decimal_places
-        self.include_sign = include_sign
-        self.mob_class = mob_class
-        self.group_with_commas = group_with_commas
-        self.digit_buff_per_font_unit = digit_buff_per_font_unit
-        self.show_ellipsis = show_ellipsis
-        self.unit = unit
-        self.unit_buff_per_font_unit = unit_buff_per_font_unit
-        self.include_background_rectangle = include_background_rectangle
-        self.edge_to_fix = edge_to_fix
-        self._font_size = font_size
-        self.fill_opacity = fill_opacity
-
-        self.initial_config = kwargs.copy()
+        self.number: float | complex = number
+        self.num_decimal_places: int = num_decimal_places
+        self.include_sign: bool = include_sign
+        self.mob_class: type[TextLike] = mob_class
+        self.group_with_commas: bool = group_with_commas
+        self.digit_buff_per_font_unit: float = digit_buff_per_font_unit
+        self.show_ellipsis: bool = show_ellipsis
+        self.unit: str | None = unit
+        self.unit_buff_per_font_unit: float = unit_buff_per_font_unit
+        self.include_background_rectangle: bool = include_background_rectangle
+        self.edge_to_fix: Vector3D = edge_to_fix
+        self._font_size: float = font_size
+        self.fill_opacity: float = fill_opacity
```

Comment:
Again, this feels a little bit suspicious - mypy should be able to figure out types from assignment?